platform/windows/display_server_windows.cpp

```diff
@@ -531,10 +531,42 @@ DisplayServer::ScreenOrientation DisplayServerWindows::screen_get_orientation(in
 }
 
 void DisplayServerWindows::screen_set_keep_on(bool p_enable) {
+	if (keep_screen_on == p_enable) {
+		return;
+	}
+
+	if (p_enable) {
+		const String reason = "Godot Engine running with display/window/energy_saving/keep_screen_on = true";
+
+		REASON_CONTEXT context;
+		context.Version = POWER_REQUEST_CONTEXT_VERSION;
+		context.Flags = POWER_REQUEST_CONTEXT_SIMPLE_STRING;
+		context.Reason.SimpleReasonString = (LPWSTR)(reason.utf16().ptrw());
```

Comment:
```suggestion
		const String reason = "Godot Engine running with display/window/energy_saving/keep_screen_on = true";
		Char16String reason_utf16 = reason.utf16();

		REASON_CONTEXT context;
		context.Version = POWER_REQUEST_CONTEXT_VERSION;
		context.Flags = POWER_REQUEST_CONTEXT_SIMPLE_STRING;
		context.Reason.SimpleReasonString = (LPWSTR)(reason_utf16.ptrw());
```

Since it's adding a pointer to the structure, not passing it directly to the function, it's important to ensure the UTF-16 version of the string is in the scope until `PowerCreateRequest` call is made, otherwise it might be deallocated before.